tempesta_fw/http_parser.c

```diff
@@ -4109,6 +4109,22 @@ tfw_http_init_parser_resp(TfwHttpResp *resp)
 			 (0x1 << TFW_HTTP_HDR_SERVER);
 }
 
+/**
+ * Adjust parser for response according to it's request.
+ */
+static void
+tfw_http_adj_parser_resp(TfwHttpResp *resp)
+{
+	TfwHttpReq *req = TFW_HTTP_RESP_BOUND_REQ(resp);
+
+	/* Can happen only in unittests. */
+	if (unlikely(!req))
+		return;
```

Comment:
We never adjust production code to satisfy tests. Tests are for production code and production code isn't for test. Please remove the check and adjust the test to handle the case.